src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,73 @@
+package mate.academy.springboot.datajpa.controller;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import mate.academy.springboot.datajpa.dto.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.ProductService;
+import mate.academy.springboot.datajpa.service.mapper.ProductMapper;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+
+    public ProductController(ProductService productService, ProductMapper productMapper) {
+        this.productService = productService;
+        this.productMapper = productMapper;
+    }
+
+    @PostMapping
+    public ProductResponseDto add(@RequestBody ProductRequestDto productRequestDto) {
+        Product product = productService
+                .add(productMapper.toModel(productRequestDto));
+        return productMapper.toDto(product);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto get(@PathVariable Long id) {
+        return productMapper.toDto(productService.get(id));
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        productService.delete(id);
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto productRequestDto) {
+        Product product = productMapper.toModel(productRequestDto);
+        product.setId(id);
+        return productMapper.toDto(productService.update(product));
+    }
+
+    @GetMapping("/byPrice")
```

Comment:
```suggestion
    @GetMapping("/by-price")
```
Let’s  use correct naming for url parameters